### ISSUE TITLE
Open header files toghether with correcponding cpp fiels.

### ISSUE THREAD

**Issue by FancyBanana** [clangd/clangd#1575]
I've noticed that very often when I open a .h file in the editor and use go to definition clangd doesn't know where to go. I have to manually open the corresponding c++ file with `alt + o` in vscode for the c++ to be added to the context.

**Comment by HighCommander4** [comment]
This usually indicates the c++ file (assuming that's what contains the definitions) is not indexed.

Does your project have a `compile_commands.json` file?

**Comment by FancyBanana** [comment]
Yes, I do have a `compile_commands.json`,  and clangd is correctly configured with it:
> I[09:41:01.925] Loaded compilation database from /home/<...>/compile_commands.json

**Comment by HighCommander4** [comment]
Does the C++ file have an entry in the `compile_commands.json` file? This is how clangd knows to index it.

If it does, then as the next diagnostic step I would suggest sharing [clangd logs](https://clangd.llvm.org/troubleshooting#gathering-logs) which may shed some light on the problem.

**Comment by zyn0217** [comment]
I guess what you want is to make Clangd index `.cpp` translation units heuristically when opening `.h` file **even** there's no corresponding TU entity in `compile_commands.json`.
We now have `switchSourceHeader` that could help to switch between source/header in some editors, which is implemented by checking if the counterpart translation unit or header exists in the same directory. But I don't know if it brings some performance degradation to make Clangd perform extra indexing every time when handling `textDocument/didOpen`.

**Comment by FancyBanana** [comment]
I've been watching the changes in `compile_commands.json` in the project and it seems that on some builds not all files are included there... When the `.cpp` file is present in the database clangd works as expected.

**Comment by FancyBanana** [comment]
I think it would be a general improvement if clangd was allowed to index an entire project directory instead of solely relying on `compile_commands.json`

**Comment by HighCommander4** [comment]
> I think it would be a general improvement if clangd was allowed to index an entire project directory instead of solely relying on `compile_commands.json`

This has been discussed in https://github.com/clangd/clangd/discussions/955
